README.md

```diff
@@ -223,6 +237,22 @@ Example of customizing semantic colors in settings.json:
 
     -   Convert import format according to `python.analysis.importFormat`.
 
+# Auto/Add import and indexing
+
+Pylance has a feature called ```auto import``` which shows symbols from installed packages and user files during completion. When committed, pylance will automatically generate an ```import``` statement. It also has something called ```add import``` which shows symbols from installed packages and user files for an unknown identifier. When selected, this will also insert an ```import``` statement.
+
+Since these features can be costly, we have enabled the minimum set and provide multiple settings to control the behavior.
+
+When ```python.analysis.indexing``` is off, we will only show symbols already used by opened files transitively. If ```indexing``` is on, we will spend some time indexing installed packages and user files when pylance is started, but after that, we will provide symbols that you haven't used yet in completion and code action.
+
+But since indexing is an expensive operation, we do not index every symbol from all installed packages but only top-level symbols. In another words, symbols in ```__all__``` in ```package/__init__```. If users want more than that, they need to use ```python.analysis.packageIndexDepths``` to tell us which packages they want us to spend more time on. Symbols indexed will show up both in completion and code actions.
+
+For user files, we index up to 2000 files. One difference between user files and installed packages is that they won't appear in completion unless they are already used. Unlike installed packages, we, by default, index all symbols from user files, which can cause thousands of user symbols to show up in completion. 
+
+We choose to show them only in code action since it requires explicit user action ```Ctrl+.``` rather than automatically triggered like completion.
```

Comment:
```suggestion
We show user file completions only in the code action since it requires an explicit user action ```Ctrl+.``` rather than being automatically triggered like completion.
```

Reply:
here as well. it is not user files, but all symbols from user files. here symbol include the file itself as a module. but also, all class and function defined in that module.

Reply:
That sounds good

Reply:
The 'them' was ambiguous to me so just wanted to fix that part.